core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -113,7 +136,7 @@ private class ShuffleStatus(numPartitions: Int) {
    * Removes all shuffle outputs associated with this host. Note that this will also remove
    * outputs which are served by an external shuffle server (if one exists).
    */
-  def removeOutputsOnHost(host: String): Unit = {
+  def removeOutputsOnHost(host: String): Unit = withWriteLock {
```

Comment:
We don't need `withWriteLock` here because `removeOutputsByFilter` is already wrapped `withWriteLock`.

Reply:
Since it's reentrant lock, I prefer to have it in all the methods in case someone adds something without guarding by lock.